recipe-server/normandy/recipes/checks.py

```diff
@@ -55,28 +57,60 @@ def recipe_signatures_are_correct(app_configs, **kwargs):
+def action_signatures_are_correct(app_configs, **kwargs):
+    errors = []
+    try:
+        Action = apps.get_model('recipes', 'Action')
+        signed_actions = list(Action.objects.exclude(signature=None))
+    except (ProgrammingError, OperationalError, ImproperlyConfigured):
+        errors.append(Info('Could not retrieve actions', id=INFO_COULD_NOT_RETRIEVE_ACTIONS))
+    else:
+        for action in signed_actions:
+            data = action.canonical_json()
+            signature = action.signature.signature
+            pubkey = action.signature.public_key
+            try:
+                signing.verify_signature(data, signature, pubkey)
+            except signing.BadSignature as e:
+                msg = f"Action '{action}' (id={action.id}) has a bad signature: {e.detail}"
+                errors.append(Warning(msg, id=WARNING_INVALID_ACTION_SIGNATURE))
+
+    return errors
+
+
+def signatures_use_good_certificates(app_configs, **kwargs):
     errors = []
     expire_early = None
     if settings.CERTIFICATES_EXPIRE_EARLY_DAYS:
         expire_early = timedelta(days=settings.CERTIFICATES_EXPIRE_EARLY_DAYS)
 
     try:
+        Signature = apps.get_model('recipes', 'Signature')
         Recipe = apps.get_model('recipes', 'Recipe')
-        signed_recipes = list(Recipe.objects.exclude(signature=None))
+        Action = apps.get_model('recipes', 'Action')
+        signatures = list(Signature.objects.all())
     except (ProgrammingError, OperationalError, ImproperlyConfigured):
-        errors.append(Info('Could not retrieve recipes', id=INFO_COULD_NOT_RETRIEVE_RECIPES))
+        errors.append(Info('Could not retrieve signatures', id=INFO_COULD_NOT_RETRIEVE_SIGNATURES))
     else:
-        urls = set(r.signature.x5u for r in signed_recipes)
+        urls = set(s.x5u for s in signatures)
         for url in urls:
             try:
                 signing.verify_x5u(url, expire_early)
             except signing.BadCertificate as exc:
                 matching_recipes = Recipe.objects.filter(signature__x5u=url)
-                count = matching_recipes.count()
-                ids = ', '.join(str(r.id) for r in matching_recipes)
-                msg = (f'{count} recipes (ids {ids}) are signed with a bad cert. {exc.detail}. '
+                matching_actions = Action.objects.filter(signature__x5u=url)
+
+                recipe_count = matching_recipes.count()
+                action_count = matching_actions.count()
+                count = recipe_count + action_count
+                msg = (f'{count} objects are signed with a bad cert. {exc.detail}. '
                        f'Certificate url is {url}')
+                if recipe_count:
+                    ids = ', '.join(str(r.id) for r in matching_recipes)
+                    msg += f'\n\tRecipe ids: {ids}'
+                if action_count:
+                    names = ', '.join(str(a.name) for a in matching_actions)
+                    msg += f'\n\tAction names: {names}'
```

Comment:
This is fine as it is, but a minor improvement might be to rely on their `__str__` representations to mark which is which, and just join the lists together at the start.